### PR TITLE
74753 initial ci pipeline for erp facade

### PR DESCRIPTION
**Pull request by amogh14893**
PR contains changes 
1.Initial CI pipeline for ERP Facade - Vulnerability checker , Unit Test and Code coverage , Build and Publish ERP Facad Wep API , ERP Facade Artifact etc.

### CODE REVIEW

**Review by jrippington — APPROVED** [review]
Looks fine to me

**Review by jrippington — APPROVED** [review]
Looks good

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,19 +1,161 @@
...
+        steps:
+          - task: UseDotNet@2
+            displayName: 'Use .NET 6.0.x sdk'
+            inputs:
+              packageType: sdk
+              useGlobalJson: true
+              workingDirectory: '$(Build.SourcesDirectory)\src'
+
+          - task: DotNetCoreCLI@2
+            displayName: ".Net Core - NuGet restore test projects only"
```
Comment (mattwin1) [inline azure-pipelines.yml:38]: This should be restore non-test projects only

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,19 +1,161 @@
...
+              noCache: true
+              nugetConfigPath: '$(Build.SourcesDirectory)\BuildNuget.config'
+              workingDirectory: '$(Build.SourcesDirectory)\src'
+              packagesDirectory: '$(Build.SourcesDirectory)\src\packagesForTests'
+
+          - task: DotNetCoreCLI@2
+            displayName: "dotnet build task"
+            inputs:
+              command: "build"
+              projects: "**/UKHO.ERPFacade.API.csproj"
```
Comment (mattwin1) [inline azure-pipelines.yml:144]: Generalise this to "**/*.csproj" please

**Review by mattwin1 — CHANGES_REQUESTED** [review]

File: azure-pipelines.yml (hunk 2 above)
Comment (amogh14893) [reply azure-pipelines.yml]: Corrected as mentioned

File: azure-pipelines.yml (hunk 1 above)
Comment (amogh14893) [reply azure-pipelines.yml:38]: > This should be restore non-test projects only

Corrcted as mentioned

**Review by mattwin1 — APPROVED** [review]
LGTM

**Review by martyn-fewtrell — APPROVED** [review]
Looks good

**Pull request merged**